src/nvim/ex_docmd.c

```diff
@@ -10107,22 +10134,16 @@ bool cmd_can_preview(char *cmd)
   if (*ea.cmd == '*') {
     ea.cmd = skipwhite(ea.cmd + 1);
   }
-  char *end = find_ex_command(&ea, NULL);
+  find_ex_command(&ea, NULL);
 
-  switch (ea.cmdidx) {
-  case CMD_substitute:
-  case CMD_smagic:
-  case CMD_snomagic:
-    // Only preview once the pattern delimiter has been typed
-    if (*end && !ASCII_ISALNUM(*end)) {
-      return true;
-    }
-    break;
-  default:
-    break;
+  if (ea.cmdidx == CMD_SIZE) {
+    return false;
+  } else if (!IS_USER_CMDIDX(ea.cmdidx)) {
+    // find_ex_command sets the flags for user commands automatically
+    ea.argt = cmdnames[(int)ea.cmdidx].cmd_argt;
```

Comment:
did `find_ex_command` have these side-effects before this PR? is that avoidable?

Reply:
It did. And I think it's fine the way it is since `find_ex_command` also has some other side-effects, so if we wanted to remove side-effects we'd have to remove all of them, which'd be quite a hassle.